packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol

```diff
@@ -53,36 +53,28 @@ library AssetLogic {
 
   /**
    * @notice Handles transferring funds from msg.sender to the Connext contract.
-   * @dev If using the native asset, will automatically wrap
+   * @dev Only allows native asset to be used for fee
    * @param _assetId - The address to transfer
    * @param _assetAmount - The specified amount to transfer. May not be the
    * actual amount transferred (i.e. fee on transfer tokens)
    * @param _fee - The fee amount in native asset included as part of the transaction that
    * should not be considered for the transfer amount.
-   * @return The assetId of the transferred asset
    */
   function handleIncomingAsset(
     address _assetId,
     uint256 _assetAmount,
     uint256 _fee
-  ) internal returns (address) {
+  ) internal {
     AppStorage storage s = LibConnextStorage.connextStorage();
 
     if (_assetId == address(0)) {
-      if (msg.value != _assetAmount + _fee) revert AssetLogic__handleIncomingAsset_notAmount();
-
-      // When transferring native asset to the contract, always make sure that the
-      // asset is properly wrapped
-      wrapNativeAsset(_assetAmount);
-      _assetId = address(s.wrapper);
-    } else {
-      if (msg.value != _fee) revert AssetLogic__handleIncomingAsset_ethWithErcTransfer();
-
-      // Transfer asset to contract
-      transferAssetToContract(_assetId, _assetAmount);
+      revert AssetLogic__handleIncomingAsset_nativeAssetNotSupported();
```

Comment:
isn't this redundant with the xcall sanity check?

Reply:
no because this isn't only used in xcall. but for xcall it is, yes

Reply:
ah hm sad

Reply:
resolved! `xcall` now calls `transferAssetToContract` directly instead of this method